internal/manager/run.go

```diff
@@ -549,6 +552,16 @@ func (m *Manager) IsReady() error {
 		if !m.synchronizer.IsReady() {
 			return errors.New("synchronizer not yet configured")
 		}
+		// Do not mark the pod ready if KIC is configured to synchronize license from Konnect but no license can be found.
+		if m.cfg.Konnect.LicenseSynchronizationEnabled {
+			licenseGetter, present := m.licenseGetter.Get()
+			if !present {
+				return errors.New("Konnect license getter not present")
+			}
+			if !licenseGetter.GetLicense().IsPresent() {
+				return errors.New("No Konnect license available")
+			}
+		}
```

Comment:
If we're not the leader we should also check the license for presence right? So that when we become the leader (old one got killed or similar) our status doesn't change suddenly.